indexer/eventprocessor.go

```diff
@@ -88,20 +98,25 @@ func (e *EventProcessor) Run() {
 			continue
 		}
 
-		headSlot := cs.Best.Slot
+		headSlot := int64(cs.Best.Slot)
 		if currentSlot >= headSlot {
 			continue
 		}
 
 		for currentSlot < headSlot {
```

Comment:
Add a check for e.ctx.Done() in this loop. A new indexer may stay in this loop for a long time.